### DISCUSSION TITLE
Grafana example is out of date

### DISCUSSION THREAD

**Discussion by csikb** [authelia/authelia#7003] [Documentation Feedback] [answered]
The Grafana dashboard is out of date. The metrics has changed since, the uuid of the dashbaord and the data source is hard coded.

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To address the issue with the Grafana dashboard being out of date due to changes in metrics, and the UUID of the dashboard and the data source being hardcoded, you will need to take a few steps. Unfortunately, as per the information provided, Authelia does not formally have any dashboards for Grafana that use Prometheus metrics for Authelia, but they welcome contributions under the Apache 2.0 license. This means you might need to manually update or create a new dashboard that reflects the current metrics used by Authelia.

1. **Review Authelia's Current Metrics**: First, review the current metrics that Authelia exports to Prometheus. This information can be found in the Authelia documentation under the metrics section. Pay special attention to the names of the metrics, the labels they use, and any new metrics that have been added or old ones that have been removed since the Grafana dashboard was last updated.

2. **Update the Grafana Dashboard JSON**: Modify the Grafana dashboard JSON to reflect the changes in the metrics. This includes updating the metric names in the panel queries and ensuring that any hardcoded UUIDs for the dashboard or data sources are either removed or updated to the correct values. If you're not the original creator of the dashboard, you might need to export the current dashboard to JSON format first.

3. **Adjust Data Source Configuration**: If the data source (Prometheus) UUID is hardcoded and has changed, you'll need to update this in the dashboard settings. It's generally recommended to use a variable for the data source when possible, to avoid such issues.

4. **Test the Dashboard**: After making the necessary changes, import the updated dashboard JSON into Grafana and verify that the metrics are being correctly displayed. Make sure to check for any errors or missing data.

5. **Contribute Back**: If you've made significant improvements or updates to the dashboard, consider contributing it back to the community. As mentioned, Authelia welcomes contributions, and sharing your updated dashboard could help others facing the same issue.

Remember, the specific steps to update the dashboard JSON will depend on the changes in the metrics and how the original dashboard was configured. If you're not familiar with Grafana dashboard JSON structure, you might need to refer to the Grafana documentation for guidance on how to properly format and update dashboard JSON files.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by csikb** [comment]
I think it would be worthwhile to add a link to the dashboard example. And update the example in the repo:
https://github.com/authelia/authelia/blob/master/examples/grafana-dashboards/simple.json

**Comment by james-d-elliott** [comment]
This is a community contributed example, additional contributions are welcomed.

**Comment by TroRon** [comment]
Does anyone already have a running example with the new metrics?

Mine is running now, but with the repo example and I can't customize it myself, because I don't know what, how and where ;)

If someone has a finished, running dashboard, I would be happy to update the repo :)

![image](https://github.com/authelia/authelia/assets/62324756/e04bc364-7aff-433e-bff2-5a1e3b16afec)

**Comment by hendrik1120** [comment, marked as answer]
I completely recreated the dashboard and fixed the outstanding issues.
https://github.com/hendrik1120/authelia/blob/grafana-dash/examples/grafana-dashboards/simple.json

It's more focused towards detecting issues and changes in the data compared to the previous example. Also features multi instance and selectable data sources. May still contain some bugs, though.

Edit: You might need to set the `scrape_interval` to 15s in order for the rate function to have enough data points to correctly approximate the curve.

**Reply by hendrik1120** [reply]
Looks like this:
![grafana](https://github.com/authelia/authelia/assets/89412959/56a2edfb-7cfe-431a-8798-16506a1ff16b)

**Reply by TroRon** [reply]
Wow perfect, I'd love to take a look tonight.

Thank you very much 😀

**Reply by TroRon** [reply]
Hi Hendrik

I have adapted it so far and it works, thanks again.

![image](https://github.com/authelia/authelia/assets/62324756/28551e9e-efc7-415d-8713-859e8a477454)


However, I have at:

Requests
Top slow requests
OIDC requests

![image](https://github.com/authelia/authelia/assets/62324756/8d14305a-5669-4fe9-8502-b822ddb9af15)
![image](https://github.com/authelia/authelia/assets/62324756/a60fecac-6f54-4c95-a4ca-33b1e975e3ad)

only “No data” in each case.

Do I have to adjust anything here so that these values are also filled?

**Reply by hendrik1120** [reply]
Did you already set the prometheus scrape_interval to 15s ?

**Reply by hendrik1120** [reply]
Also, which authelia version are you using? I built this specifically for 4.38

**Reply by TroRon** [reply]
Yes, the Scrape_Interval is set to 15 seconds.

Hmmm .... I am currently using 4.38.8, the latest version

**Reply by hendrik1120** [reply]
That's interesting. I am unable to replicate this. My configs for authelia 4.38.8, latest grafana and latest prometheus:

prometheus.yml
```yaml
global:
  scrape_interval:     15s
  evaluation_interval: 15s
  external_labels:
      monitor: 'my-project'

rule_files:
  - 'alert.rules'

scrape_configs:
  - job_name: authelia
    scheme: http
    static_configs:
    - targets: ['authelia:9959']
```

authelia configuration.yml
```yaml
telemetry:
  metrics:
    enabled: true
```

**Reply by TroRon** [reply]
Yes, that's really very exciting ...

I've compared the config, it fits so far ...

I'll keep looking ;)

Thank you :)

**Reply by TroRon** [reply]
Hi Hendrik

Sorry to have to bother you again.
I have been looking at the whole thing for the last 2 days now and have run various tests.

This is what I noticed:
![image](https://github.com/authelia/authelia/assets/62324756/065c242f-b136-40a9-b0ed-b3dfc9b6fc4c)
![image](https://github.com/authelia/authelia/assets/62324756/e340c114-5ff8-4c82-b480-7df52a5d6aa6)

Could this be related to the problem that I am not receiving any data?

**Reply by hendrik1120** [reply]
Moin, well that's because you can only use the `label_values` function in grafana templates, not in prometheus. Also, if you look at the usages of the `instance` variable, you will see that it is being used for every query on the dashboard. For that reason, none of the graphs should be showing anything.
Can you just query `authelia_request` in prometheus and show me the result?

**Reply by TroRon** [reply]
Moin Hendrik ;)

Many thanks for the answer.
Here is the answer to the requested query:
![image](https://github.com/authelia/authelia/assets/62324756/32def486-7b2f-43ff-825d-03456b8f5e3f)

Also exciting, I get answers to everything here ....

Which is actually a good thing, if I've interpreted that correctly
![image](https://github.com/authelia/authelia/assets/62324756/9f640493-b12c-4361-bb2f-b0a1880709ef)

Best regards from Switzerland

Ronny

**Reply by TroRon** [reply]
Found:

![image](https://github.com/authelia/authelia/assets/62324756/ef64cf3d-5bf5-4341-8a0f-fadfad0258de)

when I select my instance in the dropdown, data appears ....

**Reply by hendrik1120** [reply]
Then you can completely disregard my previous comment and just select your instance in the global variable at the top left of the dashboard. I will take a look at the top slow request data, I still have to check that accuracy anyway...

**Reply by hendrik1120** [reply]
The issue was that for some panels, I used the = operator to match the instance instead of the =~ operator. Which does not work if no instance is selected. I committed the fix to https://github.com/hendrik1120/authelia/blob/grafana-dash/examples/grafana-dashboards/simple.json Feel free to check it out. All panels should be working now.

**Reply by TroRon** [reply]
Hi Hendrik

Just tested it, now works as expected ;)

Thank you very much for your work and support

**Reply by james-d-elliott** [reply]
Feel free to PR this if you have not already. Thanks for fixing it up.
